### ISSUE TITLE
tests: Use loopback to find any available port

### ISSUE THREAD

**Issue by BourgondAries** [amethyst/laminar#193]
Hard-coding test ports is error-prone. We should use the OS loopback interface to find any available port instead of manually ensuring that ports on each test are different.

**Comment by TimonPost** [comment]
Agree, we did do that, however, when I did this in the past I got some 'address is already in use errors', not sure why it did not resolve the port automatically.

**Comment by BourgondAries** [comment]
I'm not sure how you implemented it but might be a TOCTOU error. (I can't find the use of loopbacks in the git log)

**Comment by TimonPost** [comment]
I guess there is one way to find out, just change the tests and see what it does.

**Comment by BourgondAries** [comment]
Will wait for https://github.com/amethyst/laminar/pull/196 to merge

**Comment by maniyar1** [comment]
Is this still something that should happen? I [went through and patched all the tests I could find](https://github.com/maniyar1/laminar/tree/test-os-assigned-ports). There is one test in `connection_manager.rs` I omitted because it used FakeSocket. All the tests seem to run on my machine, if this vaguely looks ok I can submit a PR.
